File: Models/Livro.cs
namespace Biblioteca;
using System.ComponentModel.DataAnnotations;
public class Livro 
{
    [Key]
    public int Id { get; set; }
    public string? Titulo { get; set; }
    public int AutorId { get; set; }
    public Autor? Autor { get; set; }
    public int EditoraId { get; set; }
    public Editora? Editora { get; set; }
    public int GeneroId { get; set; }
    public Genero? Genero { get; set; }
    public int CategoriaId { get; set; }
    public Categoria? Categoria { get; set; }
    public int IsbnId { get; set; }
    public ISBN? Isbn { get; set; }
    public int CatalogoId { get; set; }
    public Catalogo? Catalogo { get; set; }
    public int ExemplarId { get; set; }
    public Exemplar? Exemplar { get; set; }
    public int FuncionarioId { get; set; }
    public Funcionario? Funcionario { get; set; }
    public int Paginas {get; set; }
    public DateTime Publicacao { get; set; }
}

File: src/app/Livro.ts
import { Autor } from "./Autor";
import { Catalogo } from "./Catalogo";
import { Categoria } from "./Categoria";
import { Editora } from "./Editora";
import { Exemplar } from "./Exemplar";
import { Funcionario } from "./Funcionario";
import { Genero } from "./Genero";
import { ISBN } from "./ISBN";

export class Livro {

    id: number = 0;
    titulo: string = '';
    autorId : number = 0;
    autor: Autor | undefined;
    editoraId : number = 0;
    editora: Editora | undefined;
    generoId : number = 0;
    genero: Genero | undefined;
    categoriaId : number = 0;
    categoria: Categoria | undefined;
    isbnId : number = 0;
    isbn: ISBN | undefined;
    catalogoId : number = 0;
    catalogo: Catalogo | undefined;
    exemplarId : number = 0;
    exemplar: Exemplar| undefined;
    funcionarioId : number = 0;
    funcionario: Funcionario | undefined;
    paginas: number = 0;
    publicacao: Date = new Date();
}

